### ISSUE TITLE
BUG - [Sev-2] - [All] - [Accessibility: Focus moves to next dismiss button after dismissing What's New card]

### ISSUE THREAD

**Issue by eli-oat** [department-of-veterans-affairs/va-mobile-app#13171]
When a screen reader user on either iOS or Android, dismisses a card in the new "What's New" carousel on the Home screen, the focus automatically jumps to the dismiss button of another card. Because the focus lands on another dismiss button, the audio feedback makes it sound as though the initial dismiss action failed.

## Specs:

- **Device:** All supported iOS and Android devices
- **OS:** iOS / Android
- **User Account:** Any test user/Demo mode user with access to the Home screen
- **Accessibility State:** VoiceOver (iOS) or TalkBack (Android) enabled

## Steps to Reproduce

- Enable VoiceOver (iOS) or TalkBack (Android) on the device.
- Log in as any demo user, and ensure that the carousel is enabled in the dev menu
- Navigate to the Home screen where the "What's New" carousel is present.
- Swipe to move accessibility focus to a "What's New" card's dismiss button.
- Double-tap to activate the dismiss button.
- Listen to the screen reader output and note the focus indicator. It will land on the dismiss button of the next card in the carousel.

## Desired behavior
When a the dismiss button is focused by the screen reader the accessibility label should read aloud 'Dismiss. x of y. Button.' Where x is the index of the current card and y is the total number of cards. 

## Acceptance Criteria
- [ ] Focusing the dismiss button with a screen reader active should read aloud 'Dismiss. x of y. Button'
- [ ] Verified on both iOS (VoiceOver) and Android (TalkBack).

## Ticket Checklist
- [x] Steps to reproduce are defined
- [x] Desired behavior is added
- [x] Labels added (front-end, back-end, global, Health, relevant feature, accessibility, etc)
- [x] Estimate of 1 added (for front-end tickets)

**Comment by matthew-df** [comment]
After a discussion with @eli-oat we concluded an alternate approach to resolving this issue was to update the accessibility label of the dismiss button to 'Dismiss. x of y.' This will make it more clear that a new dismiss button has been focused without manually overriding the screen reader.
